fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AprCalculator.java

```diff
@@ -41,10 +41,29 @@ public BigDecimal calculateFrom(final PeriodFrequencyType interestPeriodFrequenc
             case YEARS:
                 defaultAnnualNominalInterestRate = interestRatePerPeriod.multiply(BigDecimal.valueOf(1));
             break;
+            case WHOLE_TERM:
+                        final BigDecimal ratePerPeriod = interestRatePerPeriod.divide(BigDecimal.valueOf(numberOfRepayments*repaymentEvery), 8, RoundingMode.HALF_UP);
```

Comment:
Why divide by 8?

Reply:
Its not divided by 8 but i rounded off the result of InterestRatePerPeriod divide by (numberOfRepayment*repayedEvery) to 8 us it was applied in another code in Loan.java "final MathContext mc = new MathContext(8, roundingMode);" So i assumed the roundoff that the system is using is 8.